tools/topology/m4/crossover.m4

```diff
@@ -0,0 +1,49 @@
+divert(-1)
+
+dnl Define macro for crossover widget
+
+dnl Crossover name)
+define(`N_CROSSOVER', `CROSSOVER'PIPELINE_ID`.'$1)
+
+dnl W_CROSSOVER(name, format, periods_sink, periods_source, kcontrol_list)
+define(`W_CROSSOVER',
+`SectionVendorTuples."'N_CROSSOVER($1)`_tuples_w" {'
+`	tokens "sof_comp_tokens"'
+`	tuples."word" {'
+`		SOF_TKN_COMP_PERIOD_SINK_COUNT'		STR($3)
+`		SOF_TKN_COMP_PERIOD_SOURCE_COUNT'	STR($4)
+`	}'
+`}'
+`SectionData."'N_CROSSOVER($1)`_data_w" {'
+`	tuples "'N_CROSSOVER($1)`_tuples_w"'
+`}'
+`SectionVendorTuples."'N_CROSSOVER($1)`_tuples_str" {'
+`	tokens "sof_comp_tokens"'
+`	tuples."string" {'
+`		SOF_TKN_COMP_FORMAT'	STR($2)
+`	}'
+`}'
+`SectionVendorTuples."'N_CROSSOVER($1)`_crossover_process_tuples_str" {'
+`	tokens "sof_process_tokens"'
+`	tuples."string" {'
+`		SOF_TKN_PROCESS_TYPE'		"CROSSOVER"
```

Comment:
don't need this token if you add UUID for it, e.g. refer to detect.m4

Reply:
After I removed this token, the error messages will appear while "./scripts/build-tools.sh -t"

ALSA lib data.c:700:(build_tuples) cannot find tuples CROSSOVER1.0_crossover_process_tuples_str
ALSA lib parser.c:285:(tplg_build) failed to check topology integrity

It sounds like process_tuples_str is still necessary?

Reply:
if you remove the whole definition of "'N_CROSSOVER($1)`_crossover_process_tuples_str" here, you need to remove the reference to this in line 34 below also.

Reply:
Thanks, that works.
And I guess it still need to define a tuple_uuid type with UUID in crossover.m4 file?
(according to https://github.com/thesofproject/sof/blob/master/tools/tplg_parser/tplg_parser.c#L1395)
Or we should just leave process->comp->type as NONE in tplg_load_process?
(according to https://github.com/thesofproject/sof/blob/master/tools/tplg_parser/tplg_parser.c#L1058)

Reply:
Ah please neglect my last comment because the answer is already in detect.m4 (after I rebased my project)

However, according to https://thesofproject.github.io/latest/developer_guides/uuid/index.html
I think something is still missing (e.g. parse uuid token) on running testbench. I will investigate more.

Reply:
@keyonjie I think the code of parsing uuid token should be ported to tools/tplg_parser from Linux topology driver
(mentioned in https://thesofproject.github.io/latest/developer_guides/uuid/index.html#linux-topology-driver)
if we need to run testbench for topologies with UUID token (and without process_type token).